src/test/debuginfo/generator-objects.rs

```diff
@@ -0,0 +1,68 @@
+// ignore-tidy-linelength
+
+// Require LLVM with DW_TAG_variant_part and a gdb that can read it.
+// min-system-llvm-version: 8.0
+// min-gdb-version: 8.2
+
+// compile-flags:-g
+
+// === GDB TESTS ===================================================================================
+
+// gdb-command:run
+// gdb-command:print b
+// gdb-check:$1 = generator_objects::main::generator {__0: 0x[...], <<variant>>: {__state: 0, 0: generator_objects::main::generator::Unresumed, 1: generator_objects::main::generator::Returned, 2: generator_objects::main::generator::Panicked, 3: generator_objects::main::generator::variant#3 {[...]}, 4: generator_objects::main::generator::variant#4 {[...]}}}
```

Comment:
Surprised that `__state` is inside `<<variant>>` - is that an artifact of the old layout representation, that I left around in debuginfo in #45225 (to avoid regressions), I wonder?
cc @michaelwoerister

Reply:
I think it's a consequence of the discriminant being a member of the "variant part". Here's the [DWARF for the generator](https://gist.github.com/tmandry/92e1368b51bea06d11da1e45218813db).

Given that this is how we handle enums it seems correct, but it might require changes to gdb/lldb to make it prettier.

Reply:
I mean, I don't think we're handling enums correctly (or rather, ideally).

Reply:
Can you be more specific? The DWARF generation seems "correct", but older GDB (and current LLDB) don't handle enum variants with fields well. Is that what you mean?

Reply:
Nevermind, I guess I have no idea what the syntax of `<<variant>>` is.